compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -132,9 +132,9 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
         then
           false // not an error, but not an expandable unrolled method
         else if
-          method.is(Deferred)
+            method.isLocal
+          || !method.isEffectivelyFinal
```

Comment:
@lihaoyi confirmed that *abstract methods* shouldn't be allowed to have `@unroll`. 

```suggestion
            method.is(Deferred)
          || method.isLocal
          || !method.isEffectivelyFinal
```

Reply:
I'm surprised that no tests were affected by this change!

Reply:
I've never seen `final abstract`, I didn't know that was even allowed?

Reply:
I believe the only place it's used is for the primitive classes in the standard library like `scala.Int`.

Reply:
Yeah ... that's really just primitives. A `final abstract` class cannot possibly be instantiated *nor* subclassed. So who cares what happens inside. It's basically an uninhabited type. ;)

Reply:
> I'm surprised that no tests were affected by this change!

@olhotak You're right, I was context switching when I made the comment. I've rechecked the diff and it everything seems correct to me.